packages/editor-ui/src/hooks/cloud.ts

```diff
@@ -0,0 +1,543 @@
+	getOpenWorkflowSettingsEventData,
+	getOutputModeChangedEventData,
+	getUpdatedWorkflowSettingsEventData,
+	getUserSavedCredentialsEventData,
+	getExecutionFinishedEventData,
+	getNodeRemovedEventData,
+	getNodeEditingFinishedEventData,
+	getExecutionStartedEventData,
+} from '@/hooks/segment';
+import { useNDVStore } from '@/stores/ndv.store';
+import { useWorkflowsStore } from '@/stores/workflows.store';
+import {
+	hooksGenerateNodesPanelEvent,
+	hooksResetNodesPanelSession,
+	nodesPanelSession,
+} from '@/hooks/utils/hooksNodesPanel';
+import type { ExternalHooks } from '@/mixins/externalHooks';
+import { useSegment } from '@/stores/segment.store';
+import type { PartialDeep } from 'type-fest';
+import type { IDataObject } from 'n8n-workflow';
+import type { INodeUi } from '@/Interface';
+
+export const n8nCloudHooks: PartialDeep<ExternalHooks> = {
+	parameterInput: {
+		mount: [
+			(meta) => {
+				if (!meta.parameter || !meta.inputFieldRef) {
+					return;
+				}
+
+				if (
+					meta.inputFieldRef.$el &&
+					(meta.parameter.name === 'resource' || meta.parameter.name === 'operation')
+				) {
+					const inputField = meta.inputFieldRef.$el.querySelector('input');
+					if (inputField?.classList?.value) {
+						inputField.classList.value = inputField.classList.value + ' data-hj-allow';
+					}
+				}
+			},
+		],
+	},
+	nodeCreatorSearchBar: {
+		mount: [
+			(meta) => {
+				if (!meta.inputRef) {
+					return;
+				}
+
+				meta.inputRef.classList.value = meta.inputRef.classList.value + ' data-hj-allow';
+			},
+		],
+	},
+	app: {
+		mount: [
+			() => {
+				hooksAddAdminIcon();
+			},
+			() => {
+				hooksAddFakeDoorFeatures();
+			},
+		],
+	},
+	nodeView: {
+		mount: [
+			() => {
+				const segmentStore = useSegment();
+				segmentStore.identify();
+			},
+			() => {
+				hooksAddAdminIcon();
+			},
+		],
+		createNodeActiveChanged: [
+			(meta) => {
+				const segmentStore = useSegment();
+				const eventData = {
+					source: meta.source,
+					nodes_panel_session_id: nodesPanelSession.sessionId,
+				};
+
+				hooksResetNodesPanelSession();
+				segmentStore.track('User opened nodes panel', eventData);
+				segmentStore.page('Cloud instance', 'Nodes panel', eventData);
+			},
+		],
+		addNodeButton: [
+			(meta) => {
+				const segmentStore = useSegment();
+				const eventData = {
+					eventName: 'User added node to workflow canvas',
+					properties: {
+						node_type: meta.nodeTypeName.split('.')[1],
+						nodes_panel_session_id: nodesPanelSession.sessionId,
+					},
+				};
+
+				segmentStore.track(eventData.eventName, eventData.properties);
+			},
+		],
+	},
+	main: {
+		routeChange: [
+			(meta) => {
+				const segmentStore = useSegment();
+				const splitPath = meta.to.path.split('/');
+				if (meta.from.path !== '/' && splitPath[1] === 'workflow') {
+					const eventData = {
+						workflow_id: splitPath[2],
+					};
+
+					segmentStore.page('Cloud instance', 'Workflow editor', eventData);
```

Comment:
[we really do want to get rid of segment..](https://linear.app/n8n/issue/ADO-1138/tech-debt-clean-up-segment)
just reached out to product on the ticket to make sure we get a sense of this